diff --git a/BUILD.bazel b/BUILD.bazel
index 20a7549..4b44a14 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -1,6 +1,9 @@
 load("//:build/rust.bzl", "rust_project")

-exports_files(["Cargo.toml"])
+exports_files([
+    "Cargo.toml",
+    "dist",
+])

 local_crates = ["//%s:charted_%s" % (path, name) for (path, name) in [
     ("crates/avatars", "avatars"),
@@ -63,4 +66,5 @@ rust_project(
         "@crate_index//:utoipa",
         "@crate_index//:validator",
     ] + local_crates,
+    external_data = [":dist"],
 )
